src/azure-cli/azure/cli/command_modules/resource/_params.py

```diff
@@ -228,6 +234,9 @@ def load_arguments(self, _):
         c.argument('handle_extended_json_format', arg_type=extended_json_format_type,
                    deprecate_info=c.deprecate(target='--handle-extended-json-format/-j'))
         c.argument('no_prompt', arg_type=no_prompt)
+        c.argument('confirm_with_what_if', arg_type=deployment_what_if_confirmation_type, is_preview=True)
```

Comment:
you need specify `min_api='***'` if this feature is only supported in new API version

Reply:
Added `min_api` constraints.